### PR TITLE
{geo}[intel/2019b] WRF v3.8.0

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Review by Micket — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2671.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/cbedfcc48816671314828f7e94fd7f08 for a full test report.

**Comment by boegel** [comment]
Installation fails on CentOS 8 because of:

```
mpiicc -DMPI2_SUPPORT -DFSEEKO64_OK  -o landread.o -c -w -O2 -xHost -ftz -fp-speculation=safe -fp-model source  -ip  -DDM_PARALLEL -DMAX_HISTORY=25 -DNMM_CORE=0 landread.c
landread.c(68): catastrophic error: cannot open source file "rpc/types.h"
  #include <rpc/types.h>
```

Just including `libtirpc` like in #11279 is not sufficient, we'll likely need to update the WRF easyblock to make sure it's picked up and a corresponding `-I` flag is used...

**Comment by boegel** [comment]
See also https://forum.mmm.ucar.edu/phpBB3/viewtopic.php?f=37&p=18647

**Comment by boegel** [comment]
Small correction: including `libtirpc` as dep fixes the include problem mentioned above as long as `libtirpc` adds `$EBROOTLIBTIRPC/include/tirpc` to `$CPATH` (see #11355), but then there are still linking errors:

```
ld: libwrflib.a(landread.o): in function `tsGetValueInt':
landread.c:(.text+0x1f7): undefined reference to `xdr_float'
ld: libwrflib.a(landread.o): in function `get_landuse_':
landread.c:(.text+0x477): undefined reference to `xdrstdio_create'
```

see also https://wrfforum.com/viewtopic.php?f=5&t=11000&sid=0c551eb74687e2b26e916f603d491836

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node101.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2680 v2 @ 2.80GHz, Python 2.7.5
See https://gist.github.com/2e9961846ea8d90ba403e3084470771c for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node356.hydra.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/0464e00d22a41170759a3db984dbc0de for a full test report.

**Review by lexming — CHANGES_REQUESTED** [review]
The download URLs do not work. It seems that downloading WRF needs registration at http://www2.mmm.ucar.edu/wrf/users/download/wrf-regist_or_download.php

Alternatively, they also have a repo in github at https://github.com/wrf-model/WRF and v3.8 is tagged there https://github.com/wrf-model/WRF/releases/tag/V3.8. But I have no idea if that source code in GH correlates with the tarballs used in this easyconfig.

**Comment by hajgato** [comment]
make aware that includes are in `tirpc/rpc`, meanwhile for glibc it was in `rpc`
https://github.com/easybuilders/easybuild-easyconfigs/pull/11355
you might need a `libtirpc` rebuild

**Comment by smoors** [comment]
@lexming @boegel  download works for me without registration if the http url is used instead of https:
http://www2.mmm.ucar.edu/wrf/src/WRFV3.8.TAR.gz

**Comment by boegel** [comment]
I've marked this as WIP, we should figure out the installation issues on CentOS 8 first imho...

**Comment by boegel** [comment]
@lexming @smoors Download works just fine for me with the `https://` URLs?

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3100.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/b8680b209e6ac8ed8ea108758c424c2a for a full test report.

**Comment by SebastianAchilles** [comment]
I see:
```
ERROR ERROR ERROR ERROR ERROR ERROR ERROR
'm4' utility not found! Can not configure.
```

File: easybuild/easyconfigs/w/WRF/WRF-3.8.0-intel-2019b-dmpar.eb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    'f93bb6dbb8b52d72f816e2f9a6815bffd536afeca8511552ec5abc4253a59346',  # WRF_parallel_build_fix.patch
+    '9f6bbf01f39adaf0c83a4bcda9719929d7779a4aaba2c186ed32b4ddcf6d0fc6',  # WRF_tests_limit-runtimes.patch
+    'd7223ffd8ff2b7729f29850bd9149b31ef01f46fb4a67b3ba72496da5455ea4c',  # WRF-3.8.0_netCDF-Fortran_separate_path.patch
+    '05afa8564ac26204e5138a06bc57b8e81bba158845d13cb5bb8b8faf993dac63',  # WRF-3.8.0_known_problems.patch
+    'afd968f807191d6ed607f0486cda7053d7e87b05da0e987f3db030a4e97126a7',  # WRF-3.8.0_libtirpc.patch
+]
+
+# csh is used by WRF install scripts
+builddependencies = [
+    ('tcsh', '6.22.02'),
```
Comment (SebastianAchilles) [inline easybuild/easyconfigs/w/WRF/WRF-3.8.0-intel-2019b-dmpar.eb:37]:
```suggestion
builddependencies = [
    ('M4', '1.4.18'),
    ('tcsh', '6.22.02'),
```

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
skl-centos-79 - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Core(TM) i7-8700 CPU @ 3.20GHz (skylake), Python 3.6.8
See https://gist.github.com/SebastianAchilles/b629dcfe46fda109c9a58e77db451996 for a full test report.

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
node300.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/smoors/2408fe170146af2bab18cdc012c12b54 for a full test report.

**Comment by boegel** [comment]
@smoors 
```
	forrtl: severe (174): SIGSEGV, segmentation fault occurred
```

Can you try running in an environment with a higher memory limit?

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in total)
node301.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/smoors/a9994e5dd4cbdd33dc4a532c3cf23d62 for a full test report.

**Comment by akesandgren** [comment]
Closing this PR, since it (only) uses deprecated toolchain versions,
see https://docs.easybuild.io/policies/toolchains .

**Pull request closed without merging**